### PR TITLE
Drop dependency to gems pending and bring in the MiqPassword

### PR DESCRIPTION
**Pull request by Ladas**
This should be just a temporary solution, since this service
should not be handling any passwords soon.

### CODE REVIEW

**Comment by agrare** [comment]
Nice @Ladas, could you move this under the topological_inventory namespace so it doesn't pollute anything that loads this gem?

**Comment by agrare** [comment]
And we should probably extract miq-password from gems-pending at some point, cc @carbonin @Fryguy

**Comment by carbonin** [comment]
I was hoping that this would actually be the push we needed to extract it into a more general purpose utility. I mentioned it to @bdunne once I think, but I'm not sure who would have the cycles to tackle it.

**Comment by Ladas** [comment]
@carbonin @agrare if I am not mistaken, we should not be handling secrets soon, so this code would go away?

**Comment by agrare** [comment]
@carbonin how 'bout https://github.com/agrare/miq_password ?

**Comment by bdunne** [comment]
@Fryguy looks like @agrare got to it before me.  @agrare we were talking about `ManageIQ::Password`, do you want to take care of the rename or do you want to do it?

**Comment by agrare** [comment]
> do you want to take care of the rename or do you want to do it?

Well you're not giving me much of a choice :laughing: I'll take care of it

**Comment by agrare** [comment]
@bdunne @Fryguy done, https://github.com/agrare/manageiq-password/

**Comment by Ladas** [comment]
Awesome @agrare , closing this one

**Comment by Ladas** [comment]
@agrare hm, I can actually use the gem here, you didn't send the PR right?

**Comment by Ladas** [comment]
@agrare ok, looks like these could be all changes needed, persistor works correctly after this

**Comment by agrare** [comment]
@Ladas I don't think the persister uses this actually you have to check an Authentication object from `-api`

```
>> Authentication.first.password
Traceback (most recent call last):
        1: from (irb):3
NameError (uninitialized constant ManageIQ::Password)
```

Looks like you just need to update the require

File: Gemfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ gemspec
 # your gem to rubygems.org.
 
 gem "inventory_refresh", :git => "https://github.com/ManageIQ/inventory_refresh", :branch => "master"
-gem "manageiq-gems-pending", :git => "https://github.com/ManageIQ/manageiq-gems-pending", :branch => "master"
+gem "manageiq-password", :git => "https://github.com/agrare/manageiq-password", :branch => "master"
```
[the remark this answers is not in the document]
Comment (bdunne) [reply Gemfile]: We should probably release v1.0

**Comment by Ladas** [comment]
@agrare hm, it fails for you? It's working for me locally. 

Yeah, persistor should not use it, but it fails without it.

**Comment by Ladas** [comment]
@agrare hm, doing this from API

``` Authentication.first.password ``` works for me

**Comment by agrare** [comment]
> Authentication.first.password doesn't work for me

Alright must just be me then

> Also just doing require 'manageiq-password' doesn't work for me, probably because it's not a proper gem?

You would have to do `require 'manageiq/password'`

**Comment by Ladas** [comment]
@agrare right, I tried those, I'll dig up further, also we'll need v2_key (and a way to generate that for prod deployment)

**Comment by agrare** [comment]
Yeah I think @carbonin has a way of generating that in the docker build and/or entrypoint.  For dev we can add a v2_key.dev and cp in `bin/setup`

**Comment by Ladas** [comment]
@agrare so it works for me with and without require, but we must have https://github.com/ManageIQ/topological_inventory-api/pull/36  and copy the v2_key to api root

**Pull request merged**